assets/pages/new-contributions-landing/components/ContributionForm.jsx

```diff
@@ -143,6 +143,40 @@ const getAmount = (props: PropTypes) =>
 
 // ----- Event handlers ----- //
 
+function openStripePopup(props: PropTypes) {
+  if (props.paymentHandlers.Stripe) {
+    openDialogBox(props.paymentHandlers.Stripe, getAmount(props), props.email);
+  }
+}
+
+const formHandlersForRecurring: {[PaymentMethod]: (props: PropTypes) => void} = {
+  PayPal: () => { /* TODO PayPal recurring */ },
+  Stripe: openStripePopup,
+  DirectDebit: (props: PropTypes) => { props.openDirectDebitPopUp(); },
+};
+
+const formHandlers: {
+  [Contrib]: {
```

Comment:
it would be cool if we could tell Flow to require us to provide keys for **all** possible values of `Contrib` - that would give us some extra type-safety when trying to call these functions

Reply:
WIP of this using type parameters here: https://github.com/guardian/support-frontend/compare/npf-refactor-and-comments...jr-add-all-payments-type?expand=1